### ISSUE TITLE
[BUG] AddRefitClient doesn't use the AuthorizationHeaderValueGetter

### ISSUE THREAD

**Issue by Elfocrash** [reactiveui/refit#689]
**Describe the bug**
If I use the AddRefitClient to register my client then the AuthorizationHeaderValueGetter isn't used.

I have the following interface and the following settings:
```c#
[Headers("Authorization: Bearer")]
public interface ISomeApi
{
    [Get("/users/{user}")]
    Task<ApiResponse<User>> GetUser(string user);
}

var refitSettings = new RefitSettings
{
    AuthorizationHeaderValueGetter = () => Task.FromResult("AuthKey")
};
```

If i call `RestService.For<ISomeApi>("https://api.test.com", refitSettings)` then my call is properly populated with the `Authorization` header having a value `Bearer AuthKey`.

However if I use the same `ISomeApi` interface using the `AddRefitClient` like this `services.AddRefitClient<ISomeApi>(refitSettings);` then the Authorization header's value is `Bearer`. The getter is never invoked to set the value.

**Expected behavior**
Both `RestService.For` and `AddRefitClient` should be utilizing the settings in the same way and the header should be populated.

**Environment**

- OS: Windows 10 Pro
- Version 10.0.17134
- Device: PC

**Comment by Styxxy** [comment]
Might be similar like #574 , because of the HttpClientFactory. To be determined though.

**Comment by Elfocrash** [comment]
After some debugging it looks like the AddRefitClient path completely ignores the Handler selection

**Comment by endeffects** [comment]
Any progress here?

**Comment by Elfocrash** [comment]
I had to do some DI hack locally to get this to work on my project but no actual fix yet. Both AuthorizationHeaderValueGetter and the other requestMessage one are not working.

**Comment by endeffects** [comment]
@onovotny Could you please have a look into this? Refit should be able to override this HttpClientFactory settings especially for authorization.

**Comment by onovotny** [comment]
PR's with related unit tests are welcome!

I've been busy and haven't had time to look into this.

**Comment by clairernovotny** [comment]
Fixed by #861
